core/dbt/main.py

```diff
@@ -727,6 +727,13 @@ def _build_test_subparser(subparsers, base_subparser):
         Store test results (failing rows) in the database
         '''
     )
+    sub.add_argument(
+        '--greedy',
+        action='store_true',
+        help='''
+        Run all tests that touch the selected models, even if they also depend on unselected models
```

Comment:
This is probably not strictly true either - if it is the foreign table in a relationships test, that test won't be run even though this model touches it.